File: 278-first-bad-version/278-first-bad-version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        
        int left = 1;
        int right = n;
        
        while(right >= left){
            int mid = left+(right-left)/2;
            
            if(isBadVersion(mid)){
                right = mid-1;
            }
            else{
                left = mid+1;
            }
        }
        
        if(isBadVersion(left)){
            return left;
        }
        return right;
        
        
    }
}

File: 278-first-bad-version/278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        
        left = 0
        right = n
        mid = (right+left)//2
        
        while(left <= right):
            mid = (right+left)//2
            
            if isBadVersion(mid):
                right = mid-1
            else:
                left = mid+1
                
        if isBadVersion(mid):
            return mid
        else:
            return mid+1